openembedded-core-mbl/scripts/signctl/README.md

```diff
@@ -248,91 +251,46 @@ The keys should be shown as generated in the Vault server output. If the `signct
+2. Execute the `sign` command
+   
 ```
-$ signctl sign-tfa --rpi-armstub8 /path/to/armstub8.bin --fip /path/to/fip2.bin --patch-rotkey --output-dir /path/to/output/dir/
+$ signctl sign --rpi-vc4-fw /path/to/armstub8.bin --fip /path/to/fip2.bin
 ```
 
-Several folders will be created in your output directory:
+The images will be signed in-place; the output path for the signed image will be the same as the input path.
 
-```
-├── armstub8_components
-│   ├── bl1_new.bin
-│   └── fip1_new.bin
-├── fip_components
-│   ├── nt-fw.bin
-│   ├── nt-fw-cert.bin
-│   ├── nt-fw-key-cert.bin
-│   ├── soc-fw.bin
-│   ├── soc-fw-cert.bin
-│   ├── soc-fw-key-cert.bin
-│   ├── tb-fw.bin
-│   ├── tb-fw-cert.bin
-│   ├── tos-fw.bin
-│   ├── tos-fw-cert.bin
-│   ├── tos-fw-extra1.bin
-│   ├── tos-fw-extra2.bin
-│   ├── tos-fw-key-cert.bin
-│   └── trusted-key-cert.bin
-├── new_certs
-│   ├── nt-fw-cert.bin
-│   ├── nt-fw-key-cert.bin
-│   ├── soc-fw-cert.bin
-│   ├── soc-fw-key-cert.bin
-│   ├── tb-fw-cert.bin
-│   ├── tos-fw-cert.bin
-│   ├── tos-fw-key-cert.bin
-│   └── trusted-key-cert.bin
-└── signed_images
-    ├── armstub8_new.bin
-    ├── fip1_new.bin
-    └── fip2.bin
-```
+## Signing NXP IMX Boot Components
 
-The signed_images directory contains the signed `armstub8.bin` and `fip2.bin`.
+As above, you can use the keys generated earlier to sign the certificates needed to verify the FIP.
 
-## Writing the Signed Components to the SD Card
+You also must to patch BL2 with the ROT public key, as TF-A uses this to verify the root of the certificate chain.
 
-Connect your SD card, containing an Mbed Linux OS Raspberry PI Image, to your pc. Ensure the partitions are unmounted.
+To sign the FIP, and patch BL2, follow the steps below:
 
-Write `fip2.bin` to a "raw partition" on the flash.
-Assuming your SD card is located at `/dev/sdc` run the following command.
+1. Locate `bl2.bin.imx` and `fip.bin` in the build deploy directory.
+2. Execute the `sign` command
 
 ```
-$ sudo dd if=signed_images/fip2.bin of=/dev/sdc bs=512 seek=2048
-1650+1 records in
-1650+1 records out
-845177 bytes (845 kB, 825 KiB) copied, 0.148319 s, 5.7 MB/s
+$ signctl sign --bl2 /path/to/bl2.bin --fip /path/to/fip.bin
 ```
 
-Mount the boot partition and copy `armstub8_new.bin` to it.
+<span class="notes">**Note:** On NXP 8M Mini EVK devices, BL2 is contained in `imx-boot-imx8mmevk-mbl-sd.bin-flash_evk` the path of which should be passed to the `--bl2` option instead of `bl2.bin.imx`.</span>
 
-```
-$ udisksctl mount -b /dev/sdc1
-Mounted /dev/sdc1 at /media/<user>/boot.
-```
-```
-$ cp signed_imgs/armstub8_new.bin /media/<user>/boot/armstub8.bin
-```
-```
-$ sync
-```
-```
-$ udisksctl unmount -b /dev/sdc1
-Unmounted /dev/sdc1.
-```
+## Create an update payload using the signed images
+
+Use the `create-update-payload` script as described in the [updating a device document][***TODO: link to update docs***].
```

Comment:
```suggestion
Use the `create-update-payload` script as described in the [updating a device document][https://os.mbed.com/docs/mbed-linux-os/latest/update/updating-an-mbl-image.html].
```
This where it will be.

Reply:
Added in 8af62db